### ISSUE TITLE
NestedScopeFunctions not working

### ISSUE THREAD

**Issue by kohnish** [detekt/detekt#7886]
## Expected Behavior
When NestedScopeFunctions is enabled, something like following should be warned.
```
fun printList() {
  val list = mutableListOf(1, 2)
  val result =
      list.apply {
        this.apply { this[0] = 1 }
            .apply {
              this.joinToString(", ")
                  .apply {
                    println("Result: $this")
                  }
            }
      }
  println(result)
}
```

## Observed Behavior
No warnings

## Steps to Reproduce
From Intellij community edition, enable all detekt rule and import a detekt.yaml (can be empty) as the user configuration, and paste the function somewhere in kotlin file

## Context
-

## Your Environment
* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable): -
* Gradle scan link (add `--scan` option when running the gradle task): -
* Operating System and version: Fedora 41 
* Link to your project (if it's a public repository): -

**Comment by BraisGabin** [comment]
I know really little about the detekt's IntelliJ plugin. But I think that it doesn't have the full analysis (type solving) enabled. That would explain this. That rule only works with type solving enabled.

**Comment by kohnish** [comment]
I didn't know a such thing existed. 
I checked with detektMain gradle task, and indeed the same config works.  

So intellij plugin doesn't do detektMain or equivalent to do the check. 
I'll open a ticket there, if it can be an option to do so.  
Thanks for the help.

**Comment by kohnish** [comment]
A such ticket already exists. :) 
https://github.com/detekt/detekt-intellij-plugin/issues/499
